src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -31,63 +35,92 @@ public ApiCompatRunner(string noWarn, (string, string)[] ignoredDifferences, boo
                             new Suppression
                             {
                                 DiagnosticId = difference.DiagnosticId,
                                 Target = difference.ReferenceId,
-                                Left = apicompatTuples.leftAssemblyRelativePath,
-                                Right = apicompatTuples.rightAssemblyRelativePath,
-                                IsBaselineSuppression = isBaselineSuppression
+                                Left = left.AssemblyId,
+                                Right = rightTuple.rightAssembly.AssemblyId,
+                                IsBaselineSuppression = _isBaselineSuppression
                             },
                             difference.DiagnosticId,
                             difference.Message);
                     }
+                    counter++;
                 }
             }
-            _queue.Clear();
+            _dict.Clear();
         }
 
         /// <summary>
         /// Queues the api compat for 2 assemblies.
         /// </summary>
-        /// <param name="leftPackagePath">Path to package containing left assembly.</param>
-        /// <param name="leftRelativePath">Relative left assembly path in package.</param>
-        /// <param name="rightPackagePath">Path to package containing right assembly.</param>
-        /// <param name="rightRelativePath">Relative right assembly path in package.</param>
-        /// <param name="assemblyName">The name of the assembly.</param>
-        /// <param name="compatibilityReason">The reason for assembly compatibilty.</param>
+        /// <param name="leftMetadataInfo">Metadata information for left assembly.</param>
+        /// <param name="rightMetdataInfo">Metadata information for right assembly.</param>
         /// <param name="header">The header for the api compat diagnostics.</param>
-        public void QueueApiCompat(string leftPackagePath, string leftRelativePath, string rightPackagePath, string rightRelativePath, string assemblyName, string compatibilityReason, string header)
+        public void QueueApiCompat(MetadataInformation leftMetadataInfo, MetadataInformation rightMetdataInfo, string header)
         {
-            _queue.Add((leftPackagePath, leftRelativePath, rightPackagePath, rightRelativePath, assemblyName, compatibilityReason, header));
+            if (_dict.ContainsKey(leftMetadataInfo))
+            {
+                _dict[leftMetadataInfo].Add((rightMetdataInfo, header));
+            }
+            else
+            {
+                _dict.Add(leftMetadataInfo, new List<(MetadataInformation rightAssembly, string header)>() { (rightMetdataInfo, header) });
+            }
+        }
+
+        internal void InitializePaths(string leftPackagePath, string rightPackagePath)
+        {
+            if (string.IsNullOrEmpty(leftPackagePath))
+                throw new ArgumentException(string.Format(Resources.InvalidPackagePath, leftPackagePath));
```

Comment:
what is the point of including the value of `leftPackagePath` and the value of `rightPackagePath` on these exceptions if it will be null or empty? I think we should instead pass the param name.